src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -274,6 +278,22 @@ internal void WorkerInitResponse(GrpcEvent initEvent)
             _workerInitTask.SetResult(true);
         }
 
+        private void ParseAndLogWorkerMetadata()
+        {
+            if (_initMessage?.WorkerMetadata == null)
```

Comment:
I see we use a global variable here. I understand init event happens only once Ideally. But I think it might be good idea to pass this instance as a parameter to this method, to prevent any future bugs.

Reply:
Yeah I was debating if I should pass it in or not, happy to do that just in case of future issues

Reply:
Done! Though it does feel uglier this way xD